bin/main.ml

```diff
@@ -1,18 +1,55 @@
 open Diffcessible
 open Cmdliner
+open ExtUnix.Specific
+
+let create_empty_diff_content () =
+  "diff --git a/Error b/Error\n--- a/Error\n+++ b/Error\n"
+
+let setup_input = function
+  | Some path ->
+      if not (Sys.file_exists path) then failwith ("File not found: " ^ path);
+      let ic = In_channel.open_bin path in
+      (Some ic, Notty_unix.Term.create (), None)
+  | None ->
+      let tty_path = ttyname Unix.stdout in
+      let tty_fd = Unix.openfile tty_path [ Unix.O_RDWR ] 0o600 in
+      let term = Notty_unix.Term.create ~input:tty_fd ~output:tty_fd () in
+      if not (Unix.isatty Unix.stdin) then (Some In_channel.stdin, term, None)
+      else (None, term, Some tty_fd)
+
+let safe_read_input ic is_stdin =
```

Comment:
I believe this follows your comment on closing only if not stdin?

Reply:
This seems to behave correctly but looks too complex to keep maintaining in my opinion. 

The more conventional pattern is the [`Fun.protect` function](https://ocaml.org/manual/5.2/api/Fun.html#VALprotect). It handles the exception in a more readable way, avoiding duplicating code. It can also combine well, for example, you could define a `with_input_fd` function with signature `file_path:string option -> (in_channel -> 'a) -> 'a`. This function would handle opening, calling the passed function and closing but nothing more.

The same pattern can be used for the `Term` value, can you also have a try?

Reply:
Use 

```
val with_open_bin : string -> ([t](https://ocaml.org/manual/5.2/api/In_channel.html#TYPEt) -> 'a) -> 'a
with_open_bin fn f opens a channel ic on file fn and returns f
    ic. After f returns, either with a value or by raising an exception, ic is guaranteed to be closed.
```

Reply:
One problem is reading from a closed file descriptor 

The goal is to eliminate fear of reading from a closed channel